### PR TITLE
Kazoo 4279 - Fix for HELP-19316

### PR DESCRIPTION
**Pull request by jamesaimonetti**
Fixes webhook re-enable when failure tracking hasn't flushed yet.

### CODE REVIEW

File: applications/webhooks/src/webhooks_disabler.erl [hunk 1]
```diff
@@ -72,6 +73,43 @@ find_failures() ->
...
+-spec flush_hooks(wh_json:objects()) -> non_neg_integer().
+flush_hooks(HookJObjs) ->
+    lists:sum(
+      [flush_failures(
+         wh_doc:account_id(HookJObj)
+         ,wh_doc:id(HookJObj)
+        )
+       || HookJObj <- HookJObjs
+      ]
+     ).
```
Comment (fenollp) [inline applications/webhooks/src/webhooks_disabler.erl]: I think you can write that in even more lines. Try putting one lexem per line?

File: applications/webhooks/src/webhooks_disabler.erl [hunk 2]
```diff
@@ -72,6 +73,43 @@ find_failures() ->
...
+
+-spec flush_failures(ne_binary()) -> non_neg_integer().
+-spec flush_failures(ne_binary(), api_binary()) -> non_neg_integer().
+flush_failures(AccountId) ->
+    flush_failures(AccountId, 'undefined').
```
Comment (fenollp) [inline applications/webhooks/src/webhooks_disabler.erl]: `flush_failures/1` is used nowhere. Why implement it?

File: applications/webhooks/src/webhooks_disabler.erl [hunk 3]
```diff
@@ -72,6 +73,43 @@ find_failures() ->
...
+flush_failures(AccountId, HookId) ->
+    FilterFun = fun(K, _V) ->
+                        maybe_remove_failure(K, AccountId, HookId)
+                end,
+    wh_cache:filter_erase_local(?CACHE_NAME
+                                ,FilterFun
+                               ).
```
Comment (fenollp) [inline applications/webhooks/src/webhooks_disabler.erl]: How about
```erlang
wh_cache
  :
    filter_erase_local
      (
       ?
          CACHE_NAME
       , FilterFun
      )
    .
```
?

File: applications/webhooks/src/webhooks_util.erl [hunk 4]
```diff
@@ -426,14 +403,14 @@ load_hook(Srv, WebHook) ->
...
     #webhook{id = hook_id(Hook)
-             ,uri = wh_json:get_value(<<"uri">>, Hook)
-             ,http_verb = http_verb(wh_json:get_value(<<"http_verb">>, Hook))
-             ,hook_event = hook_event(wh_json:get_value(<<"hook">>, Hook))
-             ,hook_id = wh_json:get_first_defined([<<"_id">>, <<"ID">>], Hook)
-             ,retries = retries(wh_json:get_integer_value(<<"retries">>, Hook, 3))
+             ,uri = kzd_webhook:uri(Hook)
+             ,http_verb = kzd_webhook:verb(Hook)
+             ,hook_event = hook_event(kzd_webhook:event(Hook))
+             ,hook_id = wh_doc:id(Hook)
```
Comment (fenollp) [inline applications/webhooks/src/webhooks_util.erl]: `wh_doc:id/1` looks for `_id` and `id`, not `ID`. (Maybe you missed that?)

File: core/whistle_couch-1.0.0/src/wh_doc.erl [hunk 5]
```diff
@@ -282,7 +282,7 @@ id(JObj) ->
     id(JObj, 'undefined').
 
 id(JObj, Default) ->
-    Id = wh_json:get_first_defined([?KEY_ID, <<"id">>], JObj, Default),
+    Id = wh_json:get_first_defined([?KEY_ID, <<"id">>, <<"ID">>], JObj, Default),
```
Comment (fenollp) [inline core/whistle_couch-1.0.0/src/wh_doc.erl]: Oh nevermind then

File: applications/webhooks/src/webhooks_disabler.erl (hunk 1 above)
Comment (jamesaimonetti) [inline applications/webhooks/src/webhooks_disabler.erl]: You love it

File: applications/webhooks/src/webhooks_disabler.erl (hunk 2 above)
Comment (jamesaimonetti) [inline applications/webhooks/src/webhooks_disabler.erl]: I was going to expose this in webhooks_maintenance so you could flush failures per account. Thanks for reminding me to expose it!

File: applications/webhooks/src/webhooks_disabler.erl (hunk 3 above)
Comment (jamesaimonetti) [inline applications/webhooks/src/webhooks_disabler.erl]: How about no

File: core/whistle_couch-1.0.0/src/wh_doc.erl (hunk 5 above)
Comment (jamesaimonetti) [inline core/whistle_couch-1.0.0/src/wh_doc.erl]: Got 'em! :)

Aren't you supposed to be flying somewhere?

**Comment by lazedo** [comment]
@jamesaimonetti so just missing the webhooks_maintenance entry for flush_failures/1 ?

**Comment by jamesaimonetti** [comment]
@lazedo should be in the latest commit 41babe3

**Comment by lazedo** [comment]
@jamesaimonetti miseed that. tks

**Pull request merged**